src/test/HostActivationTests/RuntimeProperties.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation
+{
+    public class RuntimeProperties : IClassFixture<RuntimeProperties.SharedTestState>
+    {
+        private SharedTestState sharedState;
+
+        public RuntimeProperties(RuntimeProperties.SharedTestState fixture)
+        {
+            sharedState = fixture;
+        }
+
+        [Fact]
+        public void CommonProperty_AppCanGetData()
+        {
+            var fixture = sharedState.RuntimePropertiesFixture
+                .Copy();
+
+            string name = "FX_PRODUCT_VERSION";
+            string value = fixture.RepoDirProvider.MicrosoftNETCoreAppVersion;
+
+            var dotnet = fixture.BuiltDotnet;
+            var appDll = fixture.TestProject.AppDll;
+            dotnet.Exec(appDll, name)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should().Pass()
+                .And.HaveStdErrContaining($"Property {name} = {value}")
+                .And.HaveStdOutContaining($"AppContext.GetData({name}) = {value}");
+        }
+
+        [Fact]
+        public void ConfigProperty_AppCanGetData()
+        {
+            var fixture = sharedState.RuntimePropertiesFixture
+                .Copy();
+
+            var dotnet = fixture.BuiltDotnet;
+            var appDll = fixture.TestProject.AppDll;
+            dotnet.Exec(appDll, sharedState.AppTestPropertyName)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should().Pass()
+                .And.HaveStdErrContaining($"Property {sharedState.AppTestPropertyName} = {sharedState.AppTestPropertyValue}")
+                .And.HaveStdOutContaining($"AppContext.GetData({sharedState.AppTestPropertyName}) = {sharedState.AppTestPropertyValue}");
+        }
+
+        [Fact]
+        public void DuplicateConfigProperty_AppConfigValueUsed()
```

Comment:
Add one more test which just reads the property from the framework runtime config - for completeness.